pkg/logging/filelogging_test.go

```diff
@@ -0,0 +1,144 @@
+	cases := []struct {
+		name                  string
+		assets                []asset.WritableAsset
+		cmdName               string
+		directory             string
+		expectedGenerationLog string
+	}{
+		{
+			name:                  "test empty assets list",
+			assets:                []asset.WritableAsset{},
+			cmdName:               "test",
+			directory:             "test",
+			expectedGenerationLog: "",
+		},
+		{
+			name: "test asset with one file",
+			assets: []asset.WritableAsset{
+				&installconfig.InstallConfig{
+					File: &asset.File{
+						Filename: "a.yaml",
+					},
+				},
+			},
+			cmdName:               "test install config",
+			directory:             "test",
+			expectedGenerationLog: "test install config created in the test directory",
+		},
+		{
+			name: "test asset with two files, same directory",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in the test directory",
+		},
+		{
+			name: "test asset with two files, two directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in the test and test/machines directories",
+		},
+		{
+			name: "test asset with two files, two directories, but three entries (same file twice",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("machines/b.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in the test and test/machines directories",
+		},
+		{
+			name: "test asset with three files, two directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("machines/c.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test/",
+			expectedGenerationLog: "machines created in the test and test/machines directories",
+		},
+		{
+			name: "test asset with three files, three directories",
+			assets: []asset.WritableAsset{
+				&machines.Master{
+					MachineFiles: []*asset.File{
+						getAsset("a.yaml"),
+						getAsset("machines/b.yaml"),
+						getAsset("control-plane/c.yaml"),
+					},
+				},
+			},
+			cmdName:               "machines",
+			directory:             "test",
+			expectedGenerationLog: "machines created in the test, test/control-plane and test/machines directories",
+		},
```

Comment:
```
a.yaml
machines/worker/infra/b1.yaml
machines/worker/infra/b2.yaml
machine/worker/c.yaml
machines/d.yaml
```

this is going to list `test, test/machines, test/machines/worker, test/machines/workers/infra` that is just a lot of directories. personally the goal should be to summarizes to like max 3 directories for example.